lib/internal/Magento/Framework/Model/ResourceModel/Db/AbstractDb.php

```diff
@@ -367,13 +367,48 @@ public function load(\Magento\Framework\Model\AbstractModel $object, $value, $fi
         return $this;
     }
 
+    /**
+     * Load an object for update (locks record)
+     *
+     * @param \Magento\Framework\Model\AbstractModel $object
+     * @param mixed $value
+     * @param string $field field to load by (defaults to model id)
+     * @return $this
+     */
+    public function loadForUpdate(\Magento\Framework\Model\AbstractModel $object, $value, $field = null)
```

Comment:
Could you please lock this functionality for the primary/unique keys only?

Reply:
I'm thinking:
```php
        if($field !== $this->getIdFieldName()) {
            throw new LocalizedException('Expected primary key');
        }
```
Let me know your thoughts (obviously a better error message).

Reply:
If non-primary keys are really unsafe to use, then the input arguments should probably be removed in favor of always using `$this->getIdFieldName()` internally.

Reply:
See the latest commits.

Reply:
Updated implementation to remove the use of the `$field` parameter.